libcxx/include/inplace_vector

```diff
@@ -0,0 +1,1091 @@
+    }
+  }
+  constexpr _LIBCPP_HIDE_FROM_ABI inplace_vector(inplace_vector&&) noexcept
+      /* strengthened */
+    requires(__base::__trivial_move_construct)
+  = default;
+  constexpr _LIBCPP_HIDE_FROM_ABI inplace_vector(initializer_list<_Tp> __il)
+      : inplace_vector(__il.begin(), __il.end()) {}
+
+  constexpr _LIBCPP_HIDE_FROM_ABI inplace_vector& operator=(const inplace_vector& __other) noexcept(
+      is_nothrow_copy_assignable_v<_Tp> && is_nothrow_copy_constructible_v<_Tp>) /* strengthened */ {
+    static_assert(_Capacity != 0, "Should have picked the trivial operator");
+    if (this != std::addressof(__other)) [[likely]] {
+      __small_size_type __assign = std::min(size(), __other.size());
+      std::copy(__other.data(), __other.data() + __assign, data());
+      if (__other.size() > size()) {
+        __append_with_size(__other.data() + __assign, __other.data() + __other.size(), __other.size() - __assign);
+      } else {
+        __resize_down(__other.size());
+      }
+    }
+    return *this;
+  }
+  constexpr _LIBCPP_HIDE_FROM_ABI inplace_vector& operator=(const inplace_vector&) noexcept
+      /* strengthened */
+    requires(__base::__trivial_copy_assign)
+  = default;
+  constexpr _LIBCPP_HIDE_FROM_ABI inplace_vector& operator=(inplace_vector&& __other) noexcept(
+      is_nothrow_move_assignable_v<_Tp> && is_nothrow_move_constructible_v<_Tp>) {
+    static_assert(_Capacity != 0, "Should have picked the trivial operator");
+    if (this != std::addressof(__other)) [[likely]] {
+      __small_size_type __assign = std::min(size(), __other.size());
+      std::move(__other.data(), __other.data() + __assign, data());
+      if (__other.size() > size()) {
+        __append_with_size(std::make_move_iterator(__other.data() + __assign),
+                           std::make_move_iterator(__other.data() + __other.size()),
+                           __other.size() - __assign);
+      } else {
+        __resize_down(__other.size());
+      }
+    }
+    return *this;
+  }
+  constexpr _LIBCPP_HIDE_FROM_ABI inplace_vector& operator=(inplace_vector&&) noexcept
+      /* strengthened */
+    requires(__base::__trivial_move_assign)
+  = default;
+
+  constexpr _LIBCPP_HIDE_FROM_ABI ~inplace_vector() = default;
+
+private:
+  constexpr _LIBCPP_HIDE_FROM_ABI iterator __make_iter(pointer __p) const noexcept {
+#  ifdef _LIBCPP_ABI_BOUNDED_ITERATORS
+    return std::__make_bounded_iter(__p, __wrap_iter<pointer>(__elems_ + 0), __wrap_iter<pointer>(__elems_ + __sz_));
+#  else
+    return iterator(__p);
+#  endif
+  }
+
+  constexpr _LIBCPP_HIDE_FROM_ABI const_iterator __make_iter(const_pointer __p) const noexcept {
+    return __make_iter(const_cast<pointer>(__p));
+  }
+
+  template <class _First, class _Sent>
+  constexpr _LIBCPP_HIDE_FROM_ABI void __assign_with_size(_First&& __first, _Sent&& __sent, size_t __n) {
+    if (__n > capacity()) [[unlikely]]
```

Comment:
```suggestion
    if (__n > capacity())
```
You call a `[[noreturn]]` function in this branch. The compiler already marks that us unlikely. No need to do it explicitly.